### PR TITLE
mgr: use new style config opts + add metadata

### PR DESCRIPTION
**Pull request by jcsp**

### CODE REVIEW

**Comment by jcsp** [comment]
Tested locally as far as a spinning up a vstart cluster to check the config gets are all type correct

File: src/mon/Monitor.cc [hunk 1]
```diff
@@ -3119,8 +3119,8 @@ void Monitor::handle_command(MonOpRequestRef op)
       osdmon()->osdmap.require_osd_release >= CEPH_RELEASE_LUMINOUS) {
     const auto& hdr = m->get_header();
     uint64_t size = hdr.front_len + hdr.middle_len + hdr.data_len;
-    uint64_t max =
-      g_conf->mon_client_bytes * g_conf->mon_mgr_proxy_client_bytes_ratio;
+    uint64_t max = g_conf->get_val<int64_t>("mon_client_bytes")
+                 * g_conf->get_val<double>("mon_mgr_proxy_client_bytes_ratio");
```
Comment (amitkumar50) [inline src/mon/Monitor.cc]: tab missed

File: src/common/options.cc [hunk 2]
```diff
@@ -1149,9 +1149,8 @@ std::vector<Option> get_global_options() {
     .set_default(100ul << 20)
     .set_description(""),
 
-    Option("mon_mgr_proxy_client_bytes_ratio", Option::TYPE_FLOAT, Option::LEVEL_ADVANCED)
-    .set_default(.3)
-    .set_description(""),
+    Option("mon_mgr_proxy_client_bytes_ratio", Option::TYPE_FLOAT, Option::LEVEL_DEV)
+    .set_default(.3),
```
Comment (tchaikov) [inline src/common/options.cc]: can we copy the comment of 
> ratio of mon_client_bytes that can be consumed by proxied mgr commands before we error out to client

from `legacy_config_opts.h` here?

i understand that this option is not a user-facing knob. but the comment might help developers. what do you think?

File: src/mon/Monitor.cc (hunk 1 above)
Comment (jcsp) [reply src/mon/Monitor.cc]: there's no rule against tabs in this codebase.

File: src/common/options.cc (hunk 2 above)
Comment (jcsp) [reply src/common/options.cc]: Done

**Comment by liewegas** [comment]
Something funny is giong on with mgr_initial_modules.  the initial mgrmap looks like
<pre>
{
    "epoch": 1,
    "active_gid": 0,
    "active_name": "",
    "active_addr": "-",
    "available": false,
    "standbys": [],
    "modules": [
        "\u0000estful",
        "status"
    ],
    "available_modules": []
}
</pre>

**Comment by liewegas** [comment]
also
<pre>
2017-09-07T01:03:56.452 INFO:tasks.ceph.mon.i.smithi070.stderr:/build/ceph-13.0.0-458-g18f7ea1/src/common/config.h: In function 'T get_typed_value_visitor<T>::operator()(U&) [with U = long unsigned int; typename boost::enable_if_c<(! boost::is_same<T1, T2>::value), int>::type <anonymous> = 0; T = long int]' thread 
7f420091f700 time 2017-09-07 01:03:54.395887
2017-09-07T01:03:56.520 INFO:tasks.ceph.mon.i.smithi070.stderr:/build/ceph-13.0.0-458-g18f7ea1/src/common/config.h: 331: FAILED assert("wrong type or option does not exist" == nullptr)
2017-09-07T01:03:56.543 INFO:tasks.ceph.mon.i.smithi070.stderr: ceph version 13.0.0-458-g18f7ea1 (18f7ea1d767659ab90dd5ed33451579985c7d55e) mimic (dev)
2017-09-07T01:03:56.568 INFO:tasks.ceph.mon.i.smithi070.stderr: 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x10e) [0x7f420afd397e]
2017-09-07T01:03:56.574 INFO:tasks.ceph.mon.i.smithi070.stderr: 2: (long md_config_t::get_val<long>(std::string const&) const+0x127) [0x7f420ade5ed7]
2017-09-07T01:03:56.577 INFO:tasks.ceph.mon.i.smithi070.stderr: 3: (Monitor::handle_command(boost::intrusive_ptr<MonOpRequest>)+0x1807) [0x7f420adb66c7]
2017-09-07T01:03:56.579 INFO:tasks.ceph.mon.i.smithi070.stderr: 4: (Monitor::dispatch_op(boost::intrusive_ptr<MonOpRequest>)+0x91f) [0x7f420adbc51f]
2017-09-07T01:03:56.582 INFO:tasks.ceph.mon.i.smithi070.stderr: 5: (Monitor::_ms_dispatch(Message*)+0x7b9) [0x7f420adbd819]
2017-09-07T01:03:56.585 INFO:tasks.ceph.mon.i.smithi070.stderr: 6: (Monitor::ms_dispatch(Message*)+0x23) [0x7f420ade6aa3]
2017-09-07T01:03:56.589 INFO:tasks.ceph.mon.i.smithi070.stderr: 7: (DispatchQueue::entry()+0x78b) [0x7f420b256a3b]
2017-09-07T01:03:56.594 INFO:tasks.ceph.mon.i.smithi070.stderr: 8: (DispatchQueue::DispatchThread::entry()+0xd) [0x7f420b06622d]
2017-09-07T01:03:56.597 INFO:tasks.ceph.mon.i.smithi070.stderr: 9: (()+0x8184) [0x7f4209d8f184]
2017-09-07T01:03:56.601 INFO:tasks.ceph.mon.i.smithi070.stderr: 10: (clone()+0x6d) [0x7f420865137d]
</pre>

**Comment by jcsp** [comment]
Oh, I didn't mention my rename of restful to \u0000estful?  I think it looks more hackerish :-)

Will fix...

**Comment by jcsp** [comment]
Fixed for both the mgr_initial_modules glitch and the assertion in handle_command

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-09-21_06:22:45-rados-wip-kefu-testing-2017-09-21-1013-distro-basic-mira/

**Pull request merged**